pgosupport.cmake

```diff
@@ -37,6 +37,18 @@ function(add_pgo TargetName)
     endforeach(ConfigType)
 endfunction(add_pgo)
 
+set(CLR_CMAKE_OPTDATA_PACKAGEID "optimization.PGO.CoreCLR")
+set(CLR_CMAKE_OPTDATA_PACKAGEWITHRID "optimization.${CLR_CMAKE_TARGET_OS}-${CLR_CMAKE_TARGET_ARCH}.PGO.CoreCLR")
+
+# Parse optdata package version from project.json
+file(TO_NATIVE_PATH "${CMAKE_SOURCE_DIR}/extract-from-json.py" ExtractFromJsonScript)
+file(TO_NATIVE_PATH "${CMAKE_SOURCE_DIR}/src/.nuget/optdata/project.json" OptDataProjectJsonPath)
+execute_process(
```

Comment:
This would run the command at the time the cmake generates build files (makefiles on Unix, Visual Studio project files on Windows), not at the build time. So if the /src/.nuget/optdata/project.json changes, it would not rerun the python script unless you clear the binary folder.
It seems that you probably want to do this at build time instead, so that it can regenerate the output when either the python script or the the .json file changes. If that's the case, you'll need the add_custom_command and add_custom_target combo instead of execute_process.